frontend/src/metabase/entities/questions.js

```diff
@@ -26,6 +27,62 @@ const Questions = createEntity({
   nameOne: "question",
   path: "/api/card",
 
+  /**
+   * Temporarily mock endpoints for Metrics v2
+   *
+   * Any question with `type: "metric"` will be passed to API as `type: "question"`.
+   * Same goes for any questions with a name starting with "Metric" (case-insensitive).
+   */
+  api: {
+    get: async payload => {
+      const get = GET("/api/card/:id");
+      const result = await get(payload);
+
+      if (result.name.toLowerCase().startsWith("metric")) {
+        return { ...result, type: "metric", dataset: false };
+      }
+
+      return result;
+    },
+
+    list: async payload => {
+      const get = GET("/api/card");
+      const results = await get(payload);
+
+      return results.map(result => {
+        if (result.name.toLowerCase().startsWith("metric")) {
+          return { ...result, type: "metric", dataset: false };
+        }
+
+        return result;
+      });
+    },
+
+    create: async payload => {
+      const create = POST("/api/card");
+
+      if (payload.type === "metric") {
+        const tweakedPayload = { ...payload, type: "question", dataset: false };
+        const result = await create(tweakedPayload);
+        return { ...result, type: "metric" };
```

Comment:
Why do we have to first create a tweakedPayload with a type "question", only to override it with a type "metric" in the last (return) step?

Reply:
Let's say we want to create a metric.
We will try to call this `create` endpoint with some payload which will include `type: "metric"`.
Backend won't accept this, as it's not implemented yet.
So this override will actually make FE send `type: "question"` to BE, but when the result comes back, we still want FE to believe it's a "metric", so we have to update the `type`.